src/routes/likernft/book/store.ts

```diff
@@ -35,24 +46,33 @@ router.get('/list', jwtOptionalAuth('read:nftbook'), async (req, res, next) => {
         moderatorWallets,
         ownerWallet,
         id,
+        timestamp,
       } = b;
       const isAuthorized = req.user
         && (req.user.wallet === ownerWallet || moderatorWallets.includes(req.user.wallet));
+      if (!isAuthorized) allAuthorized = false;
       const { stock, sold, prices } = parseBookSalesData(docPrices, isAuthorized);
       const result: any = {
         classId: id,
         prices,
         stock,
         shippingRates,
         defaultPaymentCurrency,
+        timestamp: timestamp.toMillis(),
       };
       if (req.user && req.user.wallet === wallet) {
         result.pendingNFTCount = pendingNFTCount;
         result.sold = sold;
       }
       return result;
     });
-    res.json({ list });
+    const nextKey = list.length < conditions.limit ? null : list[list.length - 1].timestamp;
+    if (allAuthorized) {
+      res.set('Cache-Control', 'public, max-age=60, s-maxage=60, stale-if-error=600');
+    } else {
+      res.set('Cache-Control', 'no-store');
+    }
```

Comment:
the condition should be flip, we should only cache resposne without any auth
or we can simple check for `req.user` not exist?
Since `jwtOptionalAuth` set no cache already

```suggestion
    if (!req.user) {
      res.set('Cache-Control', 'public, max-age=60, s-maxage=60, stale-if-error=600');
    }
```

Reply:
OK, I See, should like this?